### PR TITLE
Completed week 1 homework

### PR DESCRIPTION
**Pull request by mixelpixel**
Haven't solved matchingLeaves. There's some stuff I'd like to refactor, a few things I am still wrapping my head around & in the ES6 there's a couple things the linter doesn't like, but it's pretty much done :)

### CODE REVIEW

**Comment by mixelpixel** [comment]
I can follow your solution to matchingLeaves but will need to revisit this to be able to write it one my own from scratch.

**Comment by SunJieMing** [comment]
Good job making lots of commits

File: src/class.js [hunk 1]
```diff
@@ -6,9 +6,17 @@
 // for a potential password that will be compared to the `password` property.
 // Return true if the potential password matches the `password` property.  Otherwise return false.
 
+class User {
+  constructor(options) {
+    this.email = options.email;
+    this.password = options.password;
+  }
+  comparePasswords(potentialPassword) {
+    return this.password === potentialPassword;
```
Comment (SunJieMing) [inline src/class.js:15]: Nice

File: src/closure.js [hunk 2]
```diff
@@ -5,17 +5,93 @@ const counter = () => {
...
+  // version 1 √
+  let limit = 0;
+  return (...args) => {
+    limit++;
+    if (n > limit) {
+      // limit++; // <--- acts the same as limit++ before if eval
+      // console.log(limit);
+      return cb(...args);
+    }
+    return null; // <--- still not sure why we need to return null, just because the test is set up that way?
```
Comment (SunJieMing) [inline src/closure.js:68]: Yeah, we want to have something specified to return if the function has reached its limit.

**Comment by SunJieMing** [comment]
Very thorough.  Excellent work

**Comment by mixelpixel** [comment]
Thanks Ben

**Pull request closed without merging**